packages/cli/src/config/extensions/extensionSettings.ts

```diff
@@ -12,27 +12,40 @@ import { ExtensionStorage } from './storage.js';
 import type { ExtensionConfig } from '../extension.js';
 
 import prompts from 'prompts';
+import { KeychainTokenStorage } from '@google/gemini-cli-core';
 
 export interface ExtensionSetting {
   name: string;
   description: string;
   envVar: string;
+  sensitive?: boolean;
```

Comment:
Super super petty nit: Consider adding a doc block comment on this about the "default sensitivity"

Something like:

/** Whether or not the value is "sensitive".  Sensitive values will be stored in the system key chain.  If sensitive is undefined, the setting is assumed not to be sensitive */